packaging/pep517_backend/_backend.py

```diff
@@ -371,10 +372,12 @@ def get_requires_for_build_wheel(
             stacklevel=999,
         )
 
-    c_ext_build_deps = [] if is_pure_python_build else [
-        'Cython ~= 3.0.0; python_version >= "3.12"',
-        'Cython; python_version < "3.12"',
-    ]
+    if is_pure_python_build:
+        c_ext_build_deps = []
+    elif sysconfig.get_config_var('Py_GIL_DISABLED'):
+        c_ext_build_deps = ['Cython ~= 3.1.0a1']
+    else:
+        c_ext_build_deps = ['Cython ~= 3.0.11']
```

Comment:
```suggestion
        c_ext_build_deps = ['Cython ~= 3.0.12']
```

Reply:
It doesn't really matter because of how the compat operator works..